packages/iov-ethereum/src/ethereumconnection.ts

```diff
@@ -305,9 +305,9 @@ export class EthereumConnection implements BcpConnection {
     let subscriptionId: string | undefined; // ID generated by the node
     let allMessaagesSubscription: Subscription;
     const producer: Producer<BlockHeader> = {
-      start: listener => {
+      start: async listener => {
         // see https://github.com/ethereum/go-ethereum/wiki/RPC-PUB-SUB#newheads
-        this.socketSend({
+        await this.socketSend({
```

Comment:
This waits for the send completion before subscribing now. I think it is more solid to do all the `allMessaagesSubscription = socket.events.subscribe({` block before this send command.